### PR TITLE
Snfe least squares

### PR DESCRIPTION
**Pull request by keegansmith21**
Changed the flux estimation method to use a least squares approach to prevent "inf" results and minimize nonsensical answers

### CODE REVIEW

**Comment by keegansmith21** [comment]
Travis says that there's an assertion error for one of the SN tests. I've retested this and don't have the issue. Might arise due to astropy version differences? Assertion fails at the 5th decimal place so it's not a big deal.

File: tests/test_sn_flux_est.py [hunk 1]
```diff
@@ -125,31 +125,31 @@ def test_est_pulsar_sn():
     """
     I have tested this function various pulsar/observation combinations and compared to the estimated S/N from bestrof files. Here are the results:
 
-    1225462936, -b 1225462943 -e 1225463543 psr J0152-1637: The function estimated a S/N of 99.39 +/- 23.55. I compared this to what was estimated from the profile analyse_pulse_prof() and attained a S/N of 75.26 +/- 1.70
+    1225462936, -b 1225462943 -e 1225463543 psr J0152-1637: The function estimated a S/N of 100.92 +/- 22.48. I compared this to what was estimated from the profile analyse_pulse_prof() and attained a S/N of 75.26 +/- 1.70
 
-    1226062160, -b 1226062167 -e 1226062767 psr J2330-2005: Function estimated a S/N of 62.21 +/- 38.63. Compared this to the profile estimation and attained S/N of 51.51 +/- 1.16
+    1226062160, -b 1226062167 -e 1226062767 psr J2330-2005: Function estimated a S/N of 412.26 +/- 103.80. Compared this to the profile estimation and attained S/N of 51.51 +/- 1.16
```
Comment (NickSwainston) [inline tests/test_sn_flux_est.py]: Codacy Issue found: [line too long (181 > 159 characters)](https://app.codacy.com/app/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4472703)

File: tests/test_sn_flux_est.py [hunk 2]
```diff
@@ -125,31 +125,31 @@ def test_est_pulsar_sn():
...
 
-    1225713560, -b 1225713567 -e 1225714167, psr J2241-5236: Function estimated a S/N of 75.61 +/-27.27. Compared this to the profile estimation and attained S/N of 17.36 +/- 2.41
+    1225713560, -b 1225713567 -e 1225714167, psr J2241-5236: Function estimated a S/N of 119.82 +/- 41.61. Compared this to the profile estimation and attained S/N of 17.36 +/- 2.41
```
Comment (NickSwainston) [inline tests/test_sn_flux_est.py]: Codacy Issue found: [line too long (181 > 159 characters)](https://app.codacy.com/app/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4472703)

File: tests/test_sn_flux_est.py [hunk 3]
```diff
@@ -125,31 +125,31 @@ def test_est_pulsar_sn():
     """
     I have tested this function various pulsar/observation combinations and compared to the estimated S/N from bestrof files. Here are the results:
 
-    1225462936, -b 1225462943 -e 1225463543 psr J0152-1637: The function estimated a S/N of 99.39 +/- 23.55. I compared this to what was estimated from the profile analyse_pulse_prof() and attained a S/N of 75.26 +/- 1.70
+    1225462936, -b 1225462943 -e 1225463543 psr J0152-1637: The function estimated a S/N of 100.92 +/- 22.48. I compared this to what was estimated from the profile analyse_pulse_prof() and attained a S/N of 75.26 +/- 1.70
```
Comment (NickSwainston) [inline tests/test_sn_flux_est.py]: Codacy Issue found: [line too long (222 > 159 characters)](https://app.codacy.com/app/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4472703)

**Pull request merged**